### DISCUSSION TITLE
Running make command returns invalid BOARD specified no matter what

### DISCUSSION THREAD

**Discussion by ghost** [hathach/tinyusb#2893] [General]
I've been trying to get the examples described in the docs (located here: https://docs.tinyusb.org/en/latest/reference/getting_started.html) to run on my machine. I tried running the code in the dependencies section, and while the first command to run the get_deps.py for rp2040 based boards runs without giving me an error, running the next part for the get-deps fails with "invalid BOARD specified" when I supply raspberry_pi_pico. I have tried this again by supplying samd11 as the BOARD parameter, but it returned the same error. Is there something I'm doing wrong here? Looking in the hw\bsp\rp2040 directory, I do see it populated with board.h, family.c, family.h, and the boards directory has subdirectories with board.cmake files for every board that should be there. Any help on this would be greatly appreciated!

I am doing this on Windows 10

**Comment by hathach** [comment]
can you post the command that you run with the location, along with their result here.

**Comment by ghost** [comment]
![image](https://github.com/user-attachments/assets/8b7811c8-f964-4046-b2af-61524c5e80fc)

**Reply by hathach** [reply]
you only need either run the `python tools/get_deps.py rp2040`  or the make get-deps. No need todo both. Your issue is probably due to lack of pico-sdk which is not covered by get_dep script since rpi has its own doc. Please try to install pico-sdk, you may need to sest PICO_SDK_PATH env variable as well.

**Reply by ghost** [reply]
Thank you very much for your replies! I've previously had a problem with using the pico-sdk as an env variable on my machine, so I tried running it in a VSCodium project. I have the pico-sdk installed and I created it as a pico-sdk project since I figured this would be able to include it the same as if I had it as an env variable. I used the built-in git clone tool to add tinyusb into the project folder, and I believe I edited the cmakelists.txt file correctly to include tinyusb correctly, as it does compile with the changes I made. I copied the tusb_config.h from the cdc_msc example into the root directory of the project. I the ran the same python tools/get_deps.py rp2040, which seemed to work, but it still fails with the same reason in the terminal. Is my assumption that VSCodium should be able to find the pico-sdk the same as an env variable incorrect?
I don't really have experience with cmake or working with bigger projects like this, so I imagine there's a lot of knowledge I'm missing here with getting this working. I understand it if this is outside of the scope of what you can spend time helping with, but I really appreciate any help on this!

![image](https://github.com/user-attachments/assets/34763588-d18d-4835-bf57-aec367d93daf)

**Reply by hathach** [reply]
This is probably set up issue with pico-sdk or MS powershell.Can you try with the normal command prompt. You can also try to debug it a bit by injecting these command before the build_system/make/make.mk line 85 to see what is misssing e.g 

```
$(info TOP=$(TOP) CURRENT_PATH=$(CURRENT_PATH)) 
$(info FAMILY=$(FAMILY) BOARD_PATH=$(BOARD_PATH))
```

my compile log on linux machine
```
hathach@htpc:cdc_msc$ make BOARD=raspberry_pi_pico all
make -C _build/raspberry_pi_pico
make[1]: Entering directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[2]: Entering directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[3]: Entering directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[3]: Leaving directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
[  1%] Built target bs2_default
make[3]: Entering directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[3]: Leaving directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
[  4%] Built target bs2_default_library
make[3]: Entering directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[3]: Leaving directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[3]: Entering directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
[  5%] Building C object CMakeFiles/cdc_msc.dir/src/main.c.o
[  6%] Building C object CMakeFiles/cdc_msc.dir/src/msc_disk.c.o
[  7%] Building C object CMakeFiles/cdc_msc.dir/src/usb_descriptors.c.o
[  9%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/hw/bsp/rp2040/family.c.o
[ 10%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/portable/raspberrypi/rp2040/dcd_rp2040.c.o
[ 11%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/portable/raspberrypi/rp2040/rp2040_usb.c.o
[ 12%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/device/usbd.c.o
[ 13%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/device/usbd_control.c.o
[ 14%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/audio/audio_device.c.o
[ 15%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/cdc/cdc_device.c.o
[ 17%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/dfu/dfu_device.c.o
[ 18%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/dfu/dfu_rt_device.c.o
[ 19%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/hid/hid_device.c.o
[ 20%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/midi/midi_device.c.o
[ 21%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/msc/msc_device.c.o
[ 22%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/net/ecm_rndis_device.c.o
[ 23%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/net/ncm_device.c.o
[ 25%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/usbtmc/usbtmc_device.c.o
[ 26%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/vendor/vendor_device.c.o
[ 27%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/class/video/video_device.c.o
[ 28%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/portable/raspberrypi/pio_usb/dcd_pio_usb.c.o
[ 29%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/tusb.c.o
[ 30%] Building C object CMakeFiles/cdc_msc.dir/home/hathach/code/tinyusb/src/common/tusb_fifo.c.o
[ 31%] Linking CXX executable cdc_msc.elf
make[3]: Leaving directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
[100%] Built target cdc_msc
make[2]: Leaving directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
make[1]: Leaving directory '/home/hathach/code/tinyusb/examples/device/cdc_msc/_build/raspberry_pi_pico'
```

**Comment by pierrewind** [comment]
Same issue for me. First time trying to use tinyusb. 

Adding commands as suggested above does not return any BOARD_PATH or FAMILY, I can override the values but it fails shortly after.

Trying to compile for hid_inout stm32f411blackpill. Used windows 10, 11 and wsl.

_______________________________________________
Edit : got it to work with tinyusb clone in a c drive folder...

Now running into the issue of "No targets specified and no makefile found. Stop" will look into it another day.

**Reply by pierrewind** [reply]
I can confirm changing the folder of tinyusb made it work. (moved it to C: )
1) get_deps.py was not pulling the repositories desired
2) make was not finding the path of the board

Now works as expected

**Reply by hathach** [reply]
Oh I see, rule of thumbs, **DONT have SPACE** in your path. @HayleTechnica

**Reply by ghost** [reply]
I was also able to get it (mostly) working. I had to properly install the pico SDK (using the installer available from https://github.com/raspberrypi/pico-setup-windows) and only in the VS Code install that that set up was it able to find the board path. The pico-sdk plugin in VSCodium did not work, but I assume that's a problem with VSCodium. Using the installer did not allow me to run make from the regular command line, but I think it just didn't install the relevant dependencies, It's a deprecated repository now, so I guess just the regular VS Code plugin is the best way to run it. I didn't try to install the dependencies myself onto windows.

Now when I run make on an example directory I get:
![image](https://github.com/user-attachments/assets/bafdcfd2-ef34-484e-a8c7-15eeda6795ff)

Which looks like make is able to go into the proper folder as specified by the BOARD argument, it's just not finding one or both of either my project or the board folder because it hasn't been properly passed to CMake. I'm not too sure how to fix it to make that part work, but, my original problem of it not understanding what board I wanted has been fixed! thank you very much for the guidance!
